integration/test/Test/Client.hs

```diff
@@ -32,3 +41,44 @@ testClientLastActive = do
       . utcTimeToPOSIXSeconds
       <$> parseTimeM False defaultTimeLocale "%Y-%m-%dT%H:%M:%SZ" tm1
   assertBool "last_active is earlier than expected" $ ts1 >= now
+
+testListClientsIfBackendIsOffline :: HasCallStack => App ()
+testListClientsIfBackendIsOffline = do
+  resourcePool <- asks (.resourcePool)
+  ownDomain <- asString OwnDomain
+  otherDomain <- asString OtherDomain
+  [ownUser1, ownUser2] <- createAndConnectUsers [OwnDomain, OtherDomain]
+  ownClient1 <- objId $ bindResponse (API.addClient ownUser1 def) $ getJSON 201
+  ownClient2 <- objId $ bindResponse (API.addClient ownUser2 def) $ getJSON 201
+  ownUser1Id <- objId ownUser1
+  ownUser2Id <- objId ownUser2
+
+  let expectedResponse =
+        object
+          [ ownDomain .= object [ownUser1Id .= [object ["id" .= ownClient1]]],
+            otherDomain .= object [ownUser2Id .= [object ["id" .= ownClient2]]]
+          ]
+
+  bindResponse (listUsersClients ownUser1 [ownUser1, ownUser2]) $ \resp -> do
+    resp.status `shouldMatchInt` 200
+    resp.json %. "qualified_user_map" `shouldMatch` expectedResponse
+
+  runCodensity (acquireResources 1 resourcePool) $ \[downBackend] -> do
+    downUser <- runCodensity (startDynamicBackend downBackend mempty) $ \_ -> do
+      do
+        let domains = [ownDomain, otherDomain, downBackend.berDomain]
+        sequence_
+          [ API.createFedConn x (API.FedConn y "full_search")
+            | x <- domains,
+              y <- domains,
+              x /= y
+          ]
+
+      downUser <- randomUser downBackend.berDomain def
+      connectUsers ownUser1 downUser
+      connectUsers ownUser2 downUser
+      pure (downUser)
+
+    bindResponse (listUsersClients ownUser1 [ownUser1, ownUser2, downUser]) $ \resp -> do
+      resp.status `shouldMatchInt` 200
+      resp.json %. "qualified_user_map" `shouldMatch` expectedResponse
```

Comment:
Is this OK for old clients? As in what do they do if they don't see some users? Do they have code to handle this? If not we should consider adding an extra field here which indicates partial failure and doing this in the next API version.

Reply:
Before this PR, the request will return 503. After the PR, the request will return a subset of the users.
No matter how the clients are able to handle a subset of the users (and they seem to be doing it properly already), the 503 was causing a DOS anyway that would prevent calling from working completely, so we are not making things worse :)

Reply:
Alright, as far as clients don't think the users are deleted :smile:

Reply:
Maybe we should add this information to the description of the endpoint tho.